QuizApp/ViewControllers/LoginViewController.swift

```diff
@@ -257,33 +257,38 @@ class LoginViewController: UIViewController {
         
         let email = emailField.text
         let password = passwordField.text
+        guard let username = email, let password = password else { return }
         
-        let loginStatus = data.login(email: email!, password: password!)
-        //let loginStatus = LoginStatus.success
+        var success: LoginStatus?
+        let backgroundQueue = DispatchQueue(label: "login", qos: .userInitiated, attributes: .concurrent)
+        backgroundQueue.sync {
+            success = networkService.login(username: username, password: password)
+        }
```

Comment:
You don’t need to define a background queue for network requests, in fact URLSessionDataTask completion handlers are run on background thread by default, so if you need anything to run on the main queue (and in this case you do!) you need to dispatch it to the main queue manually.